h2o-core/src/main/java/water/rapids/ast/prims/assign/AstRectangleAssign.java

```diff
@@ -265,22 +235,37 @@ private void assign_frame_scalar(Frame dst, int[] cols, AstNumList rows, String
     for (int i = 0; i < cols.length; i++)
       vecs2[i] = vecs[cols[i]];
     rows.sort();                // Side-effect internal sort; needed for fast row lookup
-    new AssignFrameStringScalarTask(rows, src).doAll(vecs2);
+    AssignFrameScalarTask.doAssign(rows, vecs2, src);
   }
 
-  private static class AssignFrameStringScalarTask extends RowSliceTask {
-    private String _src;
-    private AssignFrameStringScalarTask(AstNumList rows, String src) {
+  private static class AssignFrameScalarTask extends RowSliceTask {
```

Comment:
This task assigns a constant value to a subset of rows in a vec.
There seems to be a (missed) opportunity for optimization: if a subrange of rows spans some chunks entirely, then those chunks can be replaced with the "constant value" chunks.